src/shared/get-agent-for-url.js

```diff
@@ -0,0 +1,60 @@
+// @flow
```

Comment:
This is based on https://github.com/Khan/webapp/blob/4f747e6730e41a8e6025bcf7e94b92f671cf6df0/services/graphql-gateway/src/http.js